src/dragonball/src/api/v1/virtio_net.rs

```diff
@@ -86,6 +94,7 @@ impl From<&NetworkInterfaceConfig> for VirtioNetDeviceConfigInfo {
             .unwrap_or(virtio_net_dev_mgr::DEFAULT_QUEUE_SIZE);
 
         // It is safe because we tested the type of config before.
+        #[allow(unreachable_patterns)]
         let config = match &value.backend {
```

Comment:
could we use `if let Backend::Virtio(config) = & value.backend {} else {panic!}` here to avoid `#[allow]`?

Reply:
Unfortunately, no. When enabling only `virtio-net`, cargo will complain `lf let` statement since there is one case in `Backend` enum.

```
error: irrefutable `if let` pattern
   --> src/api/v1/virtio_net.rs:103:25
    |
103 |         let config = if let Backend::Virtio(config) = & value.backend {
    |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: this pattern will always match, so the `if let` is useless
    = help: consider replacing the `if let` with a `let`
    = note: `-D irrefutable-let-patterns` implied by `-D warnings`
```